wagtail/users/models.py

```diff
@@ -43,5 +81,30 @@ def get_preferred_language(self):
     def __str__(self):
         return self.user.get_username()
 
+    @cached_property
+    def default_avatar(self):
+        return static('wagtailadmin/images/default-user-avatar.svg')
+
+    def get_avatar_url(self, size=50):
+        if self.avatar_choice == self.DEFAULT:
+            return self.default_avatar
+
+        if self.avatar_choice == self.CUSTOM:
+            try:
+                return self.avatar.url
+            except ValueError:
+                return self.default_avatar
+
+        if self.avatar_choice == self.GRAVATAR and self.user.email:
+            return get_gravatar_url(self.user.email, default=None, size=50)
+
+        return self.default_avatar
+
+    def save(self, *args, **kwargs):
+        if self.avatar:
+            this = UserProfile.objects.get(pk=self.pk)
+            this.avatar.delete(save=False)
```

Comment:
Probably it's not very critical, but there is a (small) chance that we will delete a file, but something will fail in the parent `save` method. So we will have a link to the old avatar file in the database, but we would not be able to recover a file.